### PR TITLE
Fix the `tsc` back-end task

### PR DESCRIPTION
**Pull request by juanarbol**
**Note:** EsLint is disabled intentionally; AFAIK (based on the naming structure), this is generated. It generates a lot of duplicated things in the _switch_ statement, the length of the line if very, etc., and a few more linting rules not followed

### CODE REVIEW

**Comment by RubenRBS** [comment]
Oh, I see what's going on.
https://github.com/compiler-explorer/compiler-explorer/blob/main/.eslintignore#L11 ignores js files but not ts ones. I think the whole folder should be ignored so that we can drop those eslint ignore comments :)

Can you please edit said file in the PR? :)

File: lib/asm-docs/arm32.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ export class Arm32DocumentationProvider extends BaseAssemblyDocumentationProvide
         if (!isConditionalOpcode) {
             return null;
         }
-        const information = getAsmOpcode(isConditionalOpcode[1]);
+        const information = getAsmOpcode(isConditionalOpcode[1]) as AssemblyInstructionInfo;
```
Comment (matsjla) [inline lib/asm-docs/arm32.ts:60]: This line is unnecessary because information is already typed as any because it's a JS import.

**Review by matsjla — CHANGES_REQUESTED** [review]
Hello! Could you explain what issue or problem the pull request fixes? I've tried to run the `ts-compile` and it seems to work just fine on my end.

I also suggest we don't edit any of the generated .js files because they'll just get overwritten next time they're updated.

File: lib/asm-docs/arm32.ts (hunk 1 above)
Comment (juanarbol) [reply lib/asm-docs/arm32.ts:60]: Without that casting, the command simply fails (at least on my machine)

Comment (matsjla) [reply lib/asm-docs/arm32.ts:60]: That's strange, are you running on main? `npm run ts-compile` runs fine on my machine on main

**Comment by partouf** [comment]
I'm closing this PR, I don't see any evidence of it being needed. If it still really is, I'm all ears.

**Pull request closed without merging**